packages/dart/sshnoports/bundles/universal.sh

```diff
@@ -35,435 +42,667 @@ device_atsign=""
+    string_array="$string_array\\
     <string>$1</string>"
-		shift
-	done
-	sedi "/<key>ProgramArguments<\\/key>/,/<\\/array>/c\\
+    shift
+  done
+  sedi "s|$end_line|\\
+    $end_line\\
+    |g" "$file" # make sure </array> is on a new line or we might delete something we shouldn't
+  sedi "/<key>ProgramArguments<\\/key>/,/<\\/array>/c\\
   $start_line\\
   $second_line$string_array\\
-  $end_line" "$file"
+  $end_line\\
+  " "$file"
+  sedi '/^[[:space:]]*$/d' "$file" # remove empty lines to keep things clean
+}
+
+write_systemd_user() {
+  file=$1
+  user=$2
+  sedi "s|<username>|$user|g" "$file"
 }
 
 write_systemd_environment() {
-	file=$1
-	variable=$2
-	value=$3
-	sedi "s|Environment=$variable=\".*\"|Environment=$variable=\"$value\"|g" "$file"
+  file=$1
+  variable=$2
+  value=$3
+  sedi "s|Environment=$variable=\".*\"|Environment=$variable=\"$value\"|g" "$file"
+}
+
+get_client_atsign() {
+  while [ -z "$client_atsign" ]; do
+    printf "Enter client atSign: "
+    read -r client_atsign
+  done
+}
+
+get_device_atsign() {
+  while [ -z "$device_atsign" ]; do
+    printf "Enter device atSign: "
+    read -r device_atsign
+  done
 }
 
-get_client_device_atsigns() {
-	while [ -z "$client_atsign" ]; do
-		echo "Enter client atSign:"
-		read -r client_atsign
-	done
+get_installed_atsigns() {
+  atkeycount=0
+  atkeys=""
+  selectedatsign=""
+  if [ -d "${user_home}/.atsign/keys" ]; then
+    # Disable unsafe find looping (will break if atkey file name contains a space, which it shouldn't)
+    # shellcheck disable=SC2044
+    for file in $(find "$user_home"/.atsign/keys/*.atKeys -type f); do
+      atkeys="$atkeys $(echo "$file" | sed s/^.*@// | sed s/_key.atKeys//)"
+      atkeycount=$((atkeycount + 1))
+    done
+    atkeys="$(echo "$atkeys" | sed s/\ \ /\ / | sed s/^\ //)" # remove double & leading space since it will interfere with cut
+    if [ $atkeycount -eq 0 ]; then
+      echo "$HOME/.atsign/keys directory found but there are no keys there yet"
+    elif [ $atkeycount -eq 1 ]; then
+      atkey=$(echo "$atkeys" | sed "s/\ //g")
+      echo "1 atKeys file found: ${atkey}"
+      printf "Would you like to use @%s? " "$atkey"
+      read -r use_atkey
+      case $use_atkey in
+        [Yy]*)
+          selectedatsign=$atkey
+          ;;
+      esac
+    else
+      echo "${atkeycount} atKeys found: ${atkeys}"
+      echo "Which one should be used:"
+      echo "0) None"
+      for i in $(seq 1 $atkeycount); do
+        echo "$i) @$(echo "$atkeys" | cut -d' ' -f"$i")"
+      done
+      printf '$ '
+      read -r selectedinput
+
+      selectedindex=$((selectedinput))
+      if [ "$selectedindex" -gt 0 ]; then
+        selectedatsign="$(echo "$atkeys" | cut -d' ' -f"$selectedindex")"
+        echo "Selected: @$selectedatsign"
+      else
+        echo "No existing atkeys were selected"
+      fi
+    fi
+  else
+    mkdir -p "$user_home"/.atsign/keys
+    echo "$HOME/.atsign/keys directory created"
+  fi
+}
```

Comment:
I modified this function so that it only asks y/n if a single key was found.
Otherwise it presents a numerical options list to choose from with 0/empty being don't select anything, and numbers 1+ being the various atkeys to choose from